central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -148,17 +148,30 @@ func (m *managerImpl) ProcessScanRequest(ctx context.Context, scanRequest *stora
 		// Use the created time from the DB
 		scanRequest.CreatedTime = scanConfig.GetCreatedTime()
 	}
-	err = m.scanSettingDS.UpsertScanConfiguration(ctx, scanRequest)
-	if err != nil {
-		log.Error(err)
-		return nil, errors.Errorf("Unable to save scan configuration named %q.", scanRequest.GetScanConfigName())
-	}
 
 	var profiles []string
 	for _, profile := range scanRequest.GetProfiles() {
 		profiles = append(profiles, profile.GetProfileName())
 	}
 
+	// Check if there any existing cluster that has the scan configuration with any of profiles being referenced by the scan request.
+	// If so, then we cannot create the scan configuration.
+	found, err := m.scanSettingDS.ScanConfigurationProfileExists(ctx, scanRequest.GetId(), profiles, clusters)
+
+	if err != nil {
+		log.Error(err)
+		return nil, errors.Wrapf(err, "Unable to create scan configuration named %q.", scanRequest.GetScanConfigName())
+	}
+	if found {
+		return nil, errors.Errorf("Duplicated profiles found in current or existing scan configurations: %q.", scanRequest.GetScanConfigName())
```

Comment:
Can you write a follow on ticket to update this to include the names of the conflicting scan configurations?

Reply:
https://issues.redhat.com/browse/ROX-22187